File: Iteration-2/src/TranscriptTest.java
import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.HashMap;

import org.junit.Test;

public class TranscriptTest {
    private Transcript transcript = new Transcript();

    @Test
    public void testCalculateCompleteCredit() {
        ArrayList<Course> mandatoryCourses = new ArrayList<Course>();
        mandatoryCourses.add(new MandatoryCourse("Math", "MATH101", 3, 1, "08:00-09:00", 30, 1, null));
        mandatoryCourses.add(new MandatoryCourse("English", "ENG101", 3, 1, "08:00-09:00", 30, 1, null));
        transcript.getCompletedCourses().addAll(mandatoryCourses);
        assertEquals(6, transcript.calculateCompleteCredit());
    }

    @Test
    public void testCalculateGpa() {
        HashMap<Course, Grade> mandatoryCourses = new HashMap<Course, Grade>();
        mandatoryCourses.put(new MandatoryCourse("Math", "MATH101", 3, 1, "08:00-09:00", 30, 1, null),
                new Grade(new MandatoryCourse("Math", "MATH101", 3, 1, "08:00-09:00", 30, 1, null), 100));
        transcript.getTakenCourses().putAll(mandatoryCourses);
        assertEquals(4.0, transcript.calculateGpa(), 0.0);
    }

    @Test
    public void testHasBeenPassedCourse() {
        ArrayList<Course> mandatoryCourses = new ArrayList<Course>();
        MandatoryCourse mandatoryCourse1 = new MandatoryCourse("Math", "MATH101", 3, 1, "08:00-09:00", 30, 1, null);
        mandatoryCourses.add(mandatoryCourse1);
        transcript.getCompletedCourses().addAll(mandatoryCourses);
        assertEquals(true, transcript.hasBeenPassedCourse(mandatoryCourse1));
    }

    @Test
    public void testHasBeenPassedCourses() {
        MandatoryCourse mandatoryCourse1 = new MandatoryCourse("Math", "MATH101", 3, 1, "08:00-09:00", 30, 1, null);
        ArrayList<Course> mandatoryCourses = new ArrayList<Course>();
        mandatoryCourses.add(mandatoryCourse1);
        ArrayList<Course> mandatoryCourses2 = new ArrayList<Course>();
        mandatoryCourses2.add(mandatoryCourse1);
        transcript.getCompletedCourses().addAll(mandatoryCourses);
        assertEquals(true, transcript.hasBeenPassedCourses(mandatoryCourses2));
    }

    @Test
    public void testIsCourseCompletedOrFailed() {
        MandatoryCourse mandatoryCourse1 = new MandatoryCourse("Math", "MATH101", 3, 1, "08:00-09:00", 30, 1, null);
        transcript.isCourseCompletedOrFailed(mandatoryCourse1, "AA");
        assertEquals(true, (transcript.getCompletedCourses().size() == 1));
    }
}

File: Iteration-3/Code/TranscriptTest.py
import unittest

from Transcript import Transcript
from MandatoryCourse import MandatoryCourse

class TranscriptTest(unittest.TestCase):
    
        def setUp(self) -> None:
            self.transcript = Transcript()
            self.course = MandatoryCourse("Introduction to Computer Science", "CSE1013", 4, 1, "9:00AM", 30, 1, [])
            self.course2 = MandatoryCourse("Introduction to Computer Graphics", "CSE4067", 4, 1, "11:00AM", 50, 1, [])
    
        def test_calculateCompleteCredit(self):
            self.assertEqual(self.transcript.calculateCompleteCredit(), 0)
            self.transcript.getCompletedCourses().append(self.course)
            self.assertEqual(self.transcript.calculateCompleteCredit(), 4)

        def test_isCourseCompletedOrFailed(self):
            self.assertEqual(len(self.transcript.getCompletedCourses()), 0)
            self.assertEqual(len(self.transcript.getFailedCourses()), 0)
            self.transcript.isCourseCompletedOrFailed(self.course, "AA")
            self.assertEqual(len(self.transcript.getCompletedCourses()), 1)
            self.assertEqual(len(self.transcript.getFailedCourses()), 0)
            self.transcript.isCourseCompletedOrFailed(self.course2, "FF")
            self.assertEqual(len(self.transcript.getCompletedCourses()), 1)
            self.assertEqual(len(self.transcript.getFailedCourses()), 1)

        def test_hasBeenPassed(self):
            self.transcript.isCourseCompletedOrFailed(self.course, "AA")
            self.assertEqual(self.transcript.hasBeenPassedCourses([self.course]), True)

if __name__ == '__main__':
    unittest.main()